tests/lib/tpp_schema.py

```diff
@@ -1,1057 +1,3261 @@
 
-    Patient_ID = Column(Integer, ForeignKey("Patient.Patient_ID"))
-    Patient = relationship("Patient", back_populates="ONS_CIS_New")
-    visit_date = Column(Date)
-    visit_num = Column(Integer)
-    nhs_data_share = Column(Integer)
-    last_linkage_dt = Column(Date)
-    imd_decile_E = Column(Integer)
-    imd_quartile_E = Column(Integer)
-    rural_urban = Column(Integer)
+class Therapeutics(Base):
+    __tablename__ = "Therapeutics"
+    _pk = mapped_column(t.Integer, primary_key=True)
+
+    Patient_ID = mapped_column(t.BIGINT)
+    AgeAtReceivedDate = mapped_column(t.Integer)
+    CASIM05_date_of_symptom_onset = mapped_column(t.VARCHAR(1000))
+    CASIM05_risk_cohort = mapped_column(t.VARCHAR(1000))
+    COVID_indication = mapped_column(t.VARCHAR(1000))
+    Count = mapped_column(t.Integer)
+    CurrentStatus = mapped_column(t.VARCHAR(1000))
+    Der_LoadDate = mapped_column(t.VARCHAR(1000))
+    Diagnosis = mapped_column(t.VARCHAR(1000))
+    FormName = mapped_column(t.VARCHAR(1000))
+    Intervention = mapped_column(t.VARCHAR(1000))
+    MOL1_high_risk_cohort = mapped_column(t.VARCHAR(1000))
+    MOL1_onset_of_symptoms = mapped_column(t.VARCHAR(1000))
+    Received = mapped_column(t.DateTime)
+    Region = mapped_column(t.VARCHAR(1000))
+    SOT02_onset_of_symptoms = mapped_column(t.VARCHAR(1000))
+    SOT02_risk_cohorts = mapped_column(t.VARCHAR(1000))
+    TreatmentStartDate = mapped_column(t.DateTime)
 
 
 class UKRR(Base):
     __tablename__ = "UKRR"
+    _pk = mapped_column(t.Integer, primary_key=True)
+
+    Patient_ID = mapped_column(t.BIGINT)
+    creat = mapped_column(t.REAL)
+    dataset = mapped_column(t.VARCHAR(1000))
+    eGFR_ckdepi = mapped_column(t.REAL)
+    mod_prev = mapped_column(t.VARCHAR(1000))
+    mod_start = mapped_column(t.VARCHAR(1000))
+    renal_centre = mapped_column(t.VARCHAR(1000))
+    rrt_start = mapped_column(t.Date)
+
+
+class UPRN(Base):
+    __tablename__ = "UPRN"
+    _pk = mapped_column(t.Integer, primary_key=True)
+
+    Care_Home_Flag = mapped_column(t.VARCHAR(100))
+    Pseudo_parent_uprn = mapped_column(t.VARCHAR(200))
+    Pseudo_uprn = mapped_column(t.VARCHAR(200))
+    Rural_Urban_Classification = mapped_column(t.VARCHAR(100))
+    ServiceType = mapped_column(t.VARCHAR(100))
+    Total_Pop = mapped_column(t.Integer)
+    _0to4 = mapped_column(t.Integer)
+    _10to14 = mapped_column(t.Integer)
+    _15to19 = mapped_column(t.Integer)
+    _20to24 = mapped_column(t.Integer)
+    _25to29 = mapped_column(t.Integer)
+    _30to34 = mapped_column(t.Integer)
+    _40to44 = mapped_column(t.Integer)
+    _45to49 = mapped_column(t.Integer)
+    _50to54 = mapped_column(t.Integer)
+    _55to59 = mapped_column(t.Integer)
+    _5to9 = mapped_column(t.Integer)
+    _60to64 = mapped_column(t.Integer)
+    _65to69 = mapped_column(t.Integer)
+    _70to74 = mapped_column(t.Integer)
+    _75to79 = mapped_column(t.Integer)
+    _80to84 = mapped_column(t.Integer)
+    _85Plus = mapped_column(t.Integer)
```

Comment:
thought: These columns, and maybe others, start with a single underscore which has a specific meaning when used as class attributes. I don't know if it warrants removing or replacing them.

Reply:
A fair question; I hadn't noticed these leading underscore column names. The approach I've generally taken here is to mirror the table and column names exactly without regard to any of the normal Python naming conventions. So we have class names like `ONS_Deaths` rather than `ONSDeaths`, and attribute names like `Patient_ID` rather than `patient_id`. And for names which aren't valid class or attribute names (e.g. `class`) I've just made the minimal adjustment to make them valid (e.g. `class_`).

Given that these are purely internal test classes I think the benefits of matching the database as closely as possible outweigh the downsides of having unpythonic names.

Given that the meaning of single leading underscores is just a convention I'm inclined to leave these as they are.

Reply:
Yep, that sounds fine; was just flagging in case you weren't aware and was important.